### ISSUE TITLE
Bed time not activating on galaxy watch 4

### ISSUE THREAD

**Issue by kevinkdev** [DreadedLama/dnd-sync#7]
Hi, the app is overall great, but when I try to enable bed time it does not enable on the watch. Even if I enable it on my phone, the watch only activates dnd. All permissions are granted. 

I am using a galaxy watch 4 and a pixel 7 pro running Android 14
![Screenshot_20240104_223755_sysui](https://github.com/DreadedLama/dnd-sync/assets/25504353/0babbb94-c6ac-423d-806e-3b6e34f71a72)
![Screenshot_20240104_223747_sysui](https://github.com/DreadedLama/dnd-sync/assets/25504353/2cf32dc6-b2f0-4b1a-86b8-eed7c014f3ed)
![Screenshot_20240104-223733](https://github.com/DreadedLama/dnd-sync/assets/25504353/d5f9a558-a992-4ae0-b090-c1b188b9afe7)

**Comment by DreadedLama** [comment]
I have a Galaxy Watch 4 too, tested and it works fine for me. Which version of the app are you using? Can you please try to install the apps again on both the watch and the mobile? Please use the latest v2.3 prerelease build.

**Comment by ghost** [comment]
I am using the latest v2.3 prerelease build, still no luck. DND goes on but Bed Time and Power Saving mode not.

**Comment by DreadedLama** [comment]
You are using Pixel's default Bedtime mode option right (i.e through the Digital Wellbeing app) ?

**Comment by kevinkdev** [comment]
Yes I am using the bed time mode of the pixel

**Comment by DreadedLama** [comment]
I don't have a pixel handy with me right now, but I did have one up until a couple of months ago and it was working fine. Galaxy Watch 4 with Pixel 7(on Android 14). Not sure what broke now. Can you once try the old release v2.2 and let me know if that works fine or not? Please use both the mobile and watch apps for the same.

**Comment by kevinkdev** [comment]
Used the v2.2 and got the same results. Dnd activates but bed time and power save mode no. Maybe it has to do with the fact that there was an update for the galaxy watch 4 recently?

**Comment by DreadedLama** [comment]
My Galaxy watch 4 works fine even after the recent update (but I don't have Pixel along with it). Without more info it's kinda difficult to fix this. Once I get some time I'll look into the apps to see if I can add some logging to understand your case better.

**Comment by kevinkdev** [comment]
Okay thanks, I'll try to debug it further from my side when I have time and maybe I can fix it too

**Comment by MiningMarsh** [comment]
I was having the exact same problem on my galaxy watch 6 classic, and I found what I was doing wrong.

I don't know if this is the same issue here, but I thought I'd share, as I didn't find the problem obvious:

Bedtime sync was working great for me, but I found the notification on the phone that digital wellbeing adds for bedtime mode to be very annoying. So I disabled it. I didn't realize this actually hid it from notification listeners as well, thus breaking the ability for DNDSync to detect that bedtime mode is enabled.

Re-enabling that notification on my phone restored regular functionality.

**Comment by kevinkdev** [comment]
I have checked and the notification on my phone is enabled and still not working. Anyway, thank you for sharing it!

**Comment by meazzaflavio** [comment]
Hi, I have the same issue with Google pixel 7 e galaxy watch 4, bed time mode from phone does not sync to the watch. I've checked the notification and is enabled

**Comment by cataclisma** [comment]
I have the same issue on my Galaxy Watch 4 and my Pixel 8 Pro. Notifications are on and all permissions are granted.

Any way to create a log for you for debugging?

**Comment by patheticpat** [comment]
I installed the latest release today and noticed the exact same behaviour. The problem is that the mobile app checks for a digital wellbeing notification containing the strings `on` or `paused` in the title, but if your phone is set to a language other than English, the notification title will most likely not contain those words (In German, for example, the titles are "Schlafenszeitmodus ist aktiviert" and "Schlafenszeitmodus ist pausiert"). Those strings should be localized in the app. If anybody needs a quick and dirty workaround until this is properly fixed: Set your language to English.

**Comment by Swap-File** [comment]
Thanks so much for the suggestion to make sure Digital Wellbeing notifications were not disabled.  

I had apparently never used Digital Wellbeing, so I had never given it permissions.  One click, and now DnD and bedtime mode sync works between my OnePlus12 and Galaxy Watch Ultra!

**Comment by DreadedLama** [comment]
@kevinkdev @cataclisma  @meazzaflavio yes, the issue with Bedtime sync is it's only set for the English language, as mentioned by @patheticpat. I'll be releasing a new version soon, post which the language issue would probably be fixed.

**Comment by DreadedLama** [comment]
Implemented in the last release
